GameServerLib/Logic/GameObjects/Missiles/Cone.cs

```diff
@@ -0,0 +1,181 @@
+﻿using System;
+using System.Numerics;
+using GameServerCore.Logic.Enums;
+using LeagueSandbox.GameServer.Logic.Content;
+using LeagueSandbox.GameServer.Logic.GameObjects.AttackableUnits;
+using LeagueSandbox.GameServer.Logic.GameObjects.AttackableUnits.AI;
+using LeagueSandbox.GameServer.Logic.GameObjects.AttackableUnits.Buildings.AnimatedBuildings;
+using LeagueSandbox.GameServer.Logic.GameObjects.Other;
+using LeagueSandbox.GameServer.Logic.GameObjects.Spells;
+
+
+namespace LeagueSandbox.GameServer.Logic.GameObjects.Missiles
+{
+    internal class Cone : Projectile
+    {
+        private bool _affectAsCastIsOver;
+        private float _radius;
+        private float _angleDeg;
+        private Vector2 _ownerCoords;
+        private float _beginAngle;
+        private float _endAngle;
+
+        public Cone(
+            Game game,
+            float x,
+            float y,
+            int collisionRadius,
+            AttackableUnit owner,
+            Target target,
+            Spell originSpell,
+            string nameCone,
+            int flags,
+            bool affectAsCastIsOver,
+            float angleDeg
+            ) : base(game, x, y, collisionRadius, owner, target, originSpell, 0, nameCone, flags)
+        {
+            SpellData = _game.Config.ContentManager.GetSpellData(nameCone);
+            _affectAsCastIsOver = affectAsCastIsOver;
+            _angleDeg = angleDeg;
+            CreateCone(new Target(x, y), target);            
+
+        }
+
+        public override void Update(float diff)
+        {
+            if (!_affectAsCastIsOver)
+            {
+                return;
+            }
+
+            if (_originSpell.State != SpellState.STATE_CASTING)
+            {
+                var objects = _game.ObjectManager.GetObjects().Values;
+                foreach (var obj in objects)
+                {
```

Comment:
This whole loop should be optimized in a few ways, I'll elaborate below.

1. Split the `CheckFalgsForUnit` to two different functions, one which applies the effects and one which checks if an unit is a valid target. Call the, `CheckIfValidTarget` and `ApplyEffects` for example.
2. In the loop, check unit validity _before_ measuring if said unit is within the hitbox. This saves performance, as checking flags is generally faster than doing the math required for collision check.